packages/plugins/typescript/resolvers/tests/ts-resolvers.federation.spec.ts

```diff
@@ -520,16 +546,41 @@ describe('TypeScript Resolvers Plugin + Apollo Federation', () => {
       },
     });
 
-    // UserResolver should not have a resolver function of name field
+    // `UserResolvers` should not have `username` resolver because it is marked with `@external`
+    // `UserResolvers` should have `name` resolver because whilst it is marked with `@external`, it is provided by `Book.author`
     expect(content).toBeSimilarStringTo(`
       export type UserResolvers<ContextType = any, ParentType extends ResolversParentTypes['User'] = ResolversParentTypes['User'], FederationType extends FederationTypes['User'] = FederationTypes['User']> = {
         __resolveReference?: ReferenceResolver<Maybe<ResolversTypes['User']>,
           ( { __typename: 'User' }
           & GraphQLRecursivePick<FederationType, {"id":true}> ), ContextType>;
         id?: Resolver<ResolversTypes['ID'], ParentType, ContextType>;
         name?: Resolver<Maybe<ResolversTypes['String']>, ParentType, ContextType>;
+        address?: Resolver<Maybe<ResolversTypes['Address']>, ParentType, ContextType>;
+        dateOfBirth?: Resolver<Maybe<ResolversTypes['DateOfBirth']>, ParentType, ContextType>;
+        placeOfBirth?: Resolver<Maybe<ResolversTypes['PlaceOfBirth']>, ParentType, ContextType>;
+        company?: Resolver<Maybe<ResolversTypes['Company']>, ParentType, ContextType>;
+      };
+    `);
+
+    // `AddressResolvers` should only have fields not marked with @external
+    expect(content).toBeSimilarStringTo(`
+      export type AddressResolvers<ContextType = any, ParentType extends ResolversParentTypes['Address'] = ResolversParentTypes['Address']> = {
+        zip?: Resolver<ResolversTypes['String'], ParentType, ContextType>;
       };
     `);
+
+    // `DateOfBirthResolvers` should not be generated because every field is marked with @external
+    expect(content).not.toBeSimilarStringTo('export type DateOfBirthResolvers');
+
+    // `PlaceOfBirthResolvers` should not be generated because the type is marked with @external, even if `User.placeOfBirth` is not marked with @external
+    expect(content).not.toBeSimilarStringTo('export type PlaceOfBirthResolvers');
+
+    // FIXME: `CompanyResolvers` should only have taxCode resolver because it is part of the `@provides` directive in `Book.editor`, even if the whole `Company` type is marked with @external
+    // expect(content).toBeSimilarStringTo(`
+    //   export type CompanyResolvers<ContextType = any, ParentType extends ResolversParentTypes['Company'] = ResolversParentTypes['Company']> = {
+    //     taxCode?: Resolver<ResolversTypes['String'], ParentType, ContextType>;
+    //   };
+    // `);
```

Comment:
I feel this edge case is quite hard to implement, will come back in another PR, or maybe after the major version bump.